internal/server/recon/recognize.go

```diff
@@ -123,6 +127,12 @@ func (p *placeRecognition) findPlaceCandidates(
 		return 0, nil
 	}
 
+	// Check if the first token match any abbreviated name.
+	// Note: abbreviated names are case-sensitive.
+	if places, ok := p.abbreviatedNameToPlaces[tokens[0]]; ok {
+		return 1, places
+	}
```

Comment:
To confirm, Will this work for "Sunnyvale CA" or "Texas US" ?

Reply:
Yes it works - added a test case "Sunnyvale CA" in //internal/server/v2/resolve/golden/resolve_description_test.go